src/test/mir-opt/matches_reduce_branches.rs

```diff
@@ -36,12 +36,12 @@ fn bar(i: i32) -> (bool, bool, bool, bool) {
     (a, b, c, d)
 }
 
-fn match_nested_if() -> bool {
+fn match_nested_if(flag0: bool, flag1: bool) -> bool {
     let val = match () {
-        () if if if if true { true } else { false } { true } else { false } {
-            true
+        () if if if if true { flag0 } else { flag1 } { flag1 } else { flag0 } {
+            flag0 & flag1
         } else {
-            false
+            flag0 | flag1
         } =>
```

Comment:
Sadly, this does not test `MatchBranchSimplification`. I meant something in the lines of
```rust
if if if if flag0 { true } else { false } { true } else { false } {
            true
        } else {
            false
        }
```
or additional tweaks.
The point is to keep triggering `MatchBranchSimplification` without triggering `ConstGoto`.

Reply:
I run test, it did trigger 'MatchBranchSimplification' in the last basicblock, without ConstGoto. see matches_reduce_branches.match_nested_if.MatchBranchSimplification.before/after.mir.